.github/workflows/integration_tests.yml

```diff
@@ -253,3 +397,228 @@ jobs:
+                  test_err_sum=$((test_err_sum+1))
+                fi
+              else
+                  echo ":x: Test failed (failed to compile): $filename" >> $GITHUB_STEP_SUMMARY
+                  test_err_sum=$((test_err_sum+1))
+              fi
+            fi
+          done
+          
+          # Test NVCF Python examples
+          for ex in `find examples/python -name '*.py'`; do 
+            filename=$(basename -- "$ex")
+            filename="${filename%.*}"
+            echo "Testing $filename:"
+            if [[ "$ex" == *"nvcf"* ]]; then
+              # This is an NVCF example
+              python3 $ex 1> /dev/null
+              test_status=$?
+              if [ $test_status -eq 0 ]; then
+                echo ":white_check_mark: Successfully ran test: $ex" >> $GITHUB_STEP_SUMMARY
+              else
+                echo ":x: Test failed (failed to execute): $ex" >> $GITHUB_STEP_SUMMARY
+                test_err_sum=$((test_err_sum+1))
+              fi
+            else
+              # Only run examples that are not target-specific (e.g., ionq, iqm)
+              if ! grep -q "set_target" "$ex"; then 
+                # Use --target command line option to run these examples with
+                python3 $ex --target nvcf 1> /dev/null
+                test_status=$?
+                if [ $test_status -eq 0 ]; then
+                  echo ":white_check_mark: Successfully ran test: $ex" >> $GITHUB_STEP_SUMMARY
+                else
+                  echo ":x: Test failed (failed to execute): $ex" >> $GITHUB_STEP_SUMMARY
+                  test_err_sum=$((test_err_sum+1))
+                fi
+              fi
+            fi
+          done
+          
+          set -e # Re-enable exit code error checking
+          if [ ! $test_err_sum -eq 0 ]; then
+            echo "::error::${test_err_sum} tests failed. See step summary for a list of failures"
+            exit 1
+          fi
+        shell: bash
+
+  wheel_integration_test:
+    name: Integration test using Python wheels
+    runs-on: ubuntu-latest
+    needs: [metadata, build_nvcf_image, deploy_nvcf_test_function]
+    environment: backend-validation
+    if: (success() || failure()) && (inputs.target == 'nvcf' || github.event_name == 'schedule' || inputs.target == 'nightly')
+
+    steps:
+      - name: Get code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.metadata.outputs.cudaq_commit }}
+          fetch-depth: 1
+
+      - name: Install wheel
+        # Note: we only test Python 3.10 wheel in this workflow to do a quick sanity check for NVCF integration. 
+        run: |
+          workflow_id=${{ inputs.workflow_id }}
+          if [ -z "${workflow_id}" ]; then
+            workflow_id=$(gh run list --commit ${{ needs.metadata.outputs.cudaq_commit }} --workflow Publishing --status completed --event workflow_dispatch --limit 1 --json databaseId | jq '.[0].databaseId')
```

Comment:
How long will the wheels be retained? If we don't publish any new images a week, will the oldest ones still be there?

Reply:
I've revised this logic to allow for the 'expired artifacts' condition and it seems to work when I tested it against this [run](https://github.com/NVIDIA/cuda-quantum/actions/runs/7733520121), whose artifacts are all expired.